dynamic-proxy/tests/graceful.rs

```diff
@@ -0,0 +1,62 @@
+use bytes::Bytes;
+use dynamic_proxy::body::to_simple_body;
+use dynamic_proxy::server::{HttpsConfig, SimpleHttpServer};
+use hyper::StatusCode;
+use std::convert::Infallible;
+use std::net::SocketAddr;
+use tokio::net::TcpListener;
+use tokio::time::Duration;
+
+// Ref: https://github.com/hyperium/hyper-util/blob/master/examples/server_graceful.rs
+
+async fn slow_hello_world(
+    _: hyper::Request<hyper::body::Incoming>,
+) -> Result<hyper::Response<dynamic_proxy::body::SimpleBody>, Infallible> {
+    tokio::time::sleep(Duration::from_secs(1)).await; // emulate slow request
+    let body = http_body_util::Full::<Bytes>::from("Hello, world!".to_owned());
+    let body = to_simple_body(body);
+    Ok(hyper::Response::new(body))
+}
+
+#[tokio::test]
+async fn test_graceful_shutdown() {
+    // Start the server
+    let addr = SocketAddr::from(([127, 0, 0, 1], 0));
+    let listener = TcpListener::bind(addr).await.unwrap();
+    let addr = listener.local_addr().unwrap();
+    let server = SimpleHttpServer::new(
+        hyper::service::service_fn(slow_hello_world),
+        listener,
+        HttpsConfig::Http,
+    )
+    .unwrap();
+
+    let url = format!("http://{}", addr);
+
+    // Create a client and start a POST request without finishing the body
+    let client = reqwest::Client::new();
+
+    let _client = client.clone();
+    let _url = url.clone();
+    let response_handle = tokio::spawn(async move { _client.get(&_url).send().await.unwrap() });
```

Comment:
Nit: similar comment as elsewhere about misusing the `_variable` notation, consider naming `client_cloned`, `x_cloned` instead of `_x`.